app/src/main/java/org/dhis2/usescases/splash/SplashActivity.kt

```diff
@@ -107,4 +109,13 @@ class SplashActivity : ActivityGlobalAbstract(), SplashContracts.View {
         }
     }
 
+    override fun goToNextScreen(isUserLogged: Boolean, sessionLocked: Boolean, initialSyncDone: Boolean) {
+        if (isUserLogged && initialSyncDone && !sessionLocked) {
+            startActivity(MainActivity::class.java, null, true, true, null)
+        } else if (isUserLogged && !initialSyncDone) {
+            startActivity(MainActivity::class.java, null, true, true, null)
```

Comment:
Since initialSyncDone can be either true or false and execute the same code, maybe we can omit one if clause and just check for isUserLogged and !sessionLocked?

Reply:
second branch should point to SyncActivity